test/unit/mongo/EnsureIndexesTest.php

```diff
@@ -0,0 +1,265 @@
+        $job->args = $this->args;
+        $job->job->payload['id'] = uniqid();
+        unset($job->args[$argumentName]);
+
+        $this->setExpectedException('Exception', "Argument $argumentName was not present in supplied job args for job Tripod\Mongo\Jobs\EnsureIndexes");
+        $job->perform();
+    }
+
+    /**
+     * Data provider for testMandatoryArgs
+     *
+     * @return array
+     */
+    public function mandatoryArgDataProvider()
+    {
+        return array(
+            array('tripodConfig'),
+            array('storeName'),
+            array('reindex')
+        );
+    }
+
+    /**
+     * Test the job behaves as expected
+     */
+    public function testSuccessfullyEnsureIndexesJob()
+    {
+        $job = $this->createJob();
+        $job->args = $this->createDefaultArguments();
+        $this->jobSuccessfullyEnsuresIndexes($job);
+
+        $job->perform();
+    }
+
+    /**
+     * Test that the job fails by throwing an exception
+     */
+    public function testEnsureIndexesJobThrowsErrorWhenCreatingIndexes()
+    {
+        $job = $this->createJob();
+        $job->args = $this->createDefaultArguments();
+        $this->jobThrowsExceptionWhenEnsuringIndexes($job);
+        $this->setExpectedException('Exception', "Ensuring index failed");
+
+        $job->perform();
+    }
+
+    /**
+     * test that calling the create job method on the ensureindexes job class
+     * will use the default queue name
+     */
+    public function testEnsureIndexesCreateJobDefaultQueue()
+    {
+        $jobData = array(
+            'storeName'=>'tripod_php_testing',
+            'tripodConfig'=>\Tripod\Mongo\Config::getConfig(),
+            'reindex'=>false
+        );
+
+        //create mock job
+        $job = $this->createJob();
+        $job->expects($this->once())
+            ->method('submitJob')
+            ->with(
+                \Tripod\Mongo\Config::getEnsureIndexesQueueName(),
+                'MockEnsureIndexes',
+                $jobData
+            );
+
+        $job->createJob('tripod_php_testing', false);
+    }
+
+    /**
+     * test that calling the create job method on the ensureindexes job class
+     * will throw the expected exception if redis is unreachable
+     */
+    public function testEnsureIndexesCreateJobUnreachableRedis()
+    {
+        $jobData = array(
+            'storeName'=>'tripod_php_testing',
+            'tripodConfig'=>\Tripod\Mongo\Config::getConfig(),
+            'reindex'=>false
+        );
+
+        //create mock job
+        $job = $this->getMockBuilder('\Tripod\Mongo\Jobs\EnsureIndexes')
+            ->setMethods(array('warningLog', 'enqueue'))
+            ->getMock();
+
+        $e = new Exception("Connection to Redis failed after 1 failures.Last Error : (0) php_network_getaddresses: getaddrinfo failed: nodename nor servname provided, or not known");
+
+        $job->expects($this->any())->method("enqueue")->will($this->throwException($e));
```

Comment:
Do you want to be more specific over the number of times this mocked method will be called (same goes for the other ones below)